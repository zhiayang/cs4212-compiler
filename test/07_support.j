// 07_support.j

class Main
{
	Void main()
	{
		println(new Foo().foo(420, 69));
		println(new Foo().bar("aaa", "aaa"));
		println(new Foo().bar("aaa", "a"));
		println(new Foo().bar("xxx", "bbb"));
	}
}

class Foo
{
	Int foo(Int x, Int y)
	{
		Int a; Int b; Int c; Int d; Int e; Int f; Int g; Int h;

		a = x / y;
		b = -x / y;
		c = x / -y;
		d = -x / -y;

		println(a);
		println(b);
		println(c);
		println(d);

		return 420;
	}

	Int bar(String a, String b)
	{
		println(a == b);
		println(a != b);

		return 69;
	}
}
