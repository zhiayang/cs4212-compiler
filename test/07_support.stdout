6
-6
-6
6
420
true
false
69
false
true
69
false
true
69
